qemu/tests/migration_with_sgx.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+from virttest import error_context
+from provider.sgx import SGXTestCapability
+from virttest import env_process
+from virttest.utils_misc import verify_dmesg
+from provider.sgx import SGXChecker
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Do migration test with sgx enabled
+
+    1. Check sgx and numa capability
+    2. Boot src sgx guest with numa
+    3. Migration
+    4. check sgx epc size and policy
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    sgx_cap = SGXTestCapability(test, params)
+    host_epc_size = sgx_cap.valid_host_sgx_cap()
+    params["host_epc_size"] = host_epc_size
+    if "mem" in params["mem_devs"]:
+        sgx_cap.valid_host_numa_node()
+
+    params['start_vm'] = "yes"
+    vm_name = params["vms"]
+    error_context.context("Start to boot sgx source vm", logging.info)
```

Comment:
In pullreq 3507, you use test.log.info instead of logging.info. I suggest that they be consistent in tp-qemu cases. Using test.log,  don't need to import logging. Thanks.

Reply:
Thanks Xueqiang, I updated the previous patches changes to this patch.

Reply:
yep, actually I am afraid logging.info doesn't work, we can either use test.log if we have test object, or we need to get it with logging.getLogger('avocado.test').